### PR TITLE
Update contract structure

### PR DESCRIPTION
**Pull request by shyam-patel-kira**
- Update proposal struct
- add vote and evidence structs

### CODE REVIEW

File: apps/contracts/contracts/Zengo.sol [hunk 1]
```diff
@@ -15,13 +15,33 @@ contract ZengoDAO is PermissionsEnumerable, ContractMetadata {
...
         bool isVerified;
-        uint256 votes;
+    }
+
+    struct Vote {
+        uint256 proposalId;
+        uint256 totalVotes;
+        mapping (address => bool) vote; 
+        Evidence[] evidences;
+        string[] options; // should this be enums?
```
Comment (Markkos89) [inline apps/contracts/contracts/Zengo.sol:37]: > should this be enums?

For the sake of provide my input, this answer (yes or no) relies on if the FE will pass the options for each Iteration Voting.
Defining the enums will require more effort and perhaps we can define them along the way organically, and then the contract can be updated?
Also using this array of strings give us more flexibility at this point.

just my opinion

File: apps/contracts/contracts/Zengo.sol [hunk 2]
```diff
@@ -15,13 +15,33 @@ contract ZengoDAO is PermissionsEnumerable, ContractMetadata {
...
+        Vote[] votingIterations;
+        // requires funding
+        bool isEligibleForFunding;
         bool isVerified;
-        uint256 votes;
+    }
+
+    struct Vote {
+        uint256 proposalId;
+        uint256 totalVotes;
```
Comment (Markkos89) [inline apps/contracts/contracts/Zengo.sol:34]: the `totalVotes` represents the total votes that will be distributed among the moderators?

File: apps/contracts/contracts/Zengo.sol [hunk 3]
```diff
@@ -15,13 +15,33 @@ contract ZengoDAO is PermissionsEnumerable, ContractMetadata {
...
+        mapping (address => bool) vote;
```
Comment (Markkos89) [inline apps/contracts/contracts/Zengo.sol:35]: this mapping is handling the votes, right?
having something like:

```
// other fields
uint256[] votes;
uint256[] votesWeight;
string[] options;
```
where the votes are the votes count on eachoption 
and the votesWeight is the weight the vote represented to that option.
and with this data apply an `sqrt` math method to calculate winner/s ?

just asking :grinning:

File: apps/contracts/contracts/Zengo.sol [hunk 4]
```diff
@@ -77,24 +97,35 @@ contract ZengoDAO is PermissionsEnumerable, ContractMetadata {
 
     function submitProposal(
         string calldata _title,
-        string calldata _description
+        string calldata _proposalDescription,
+        string calldata _evidenceDescription,
+        uint256 _latitude,
+        uint256 _longitude,
+        string _proposalType,
+        string _streetAddress
```
Comment (Markkos89) [inline apps/contracts/contracts/Zengo.sol:105]: these are the required fields to submit a proposal?

```
title
proposalType
proposalDescription
evidenceDate
evidenceDescription
evidenceIpfsUrl // the evidence file is beeing uploaded to IPFS and we need to to store the url, right?
streetAddress
latitude
longitude
 ```
   
  just asking :smile: 
  Hope it helps

**Review by Markkos89 — COMMENTED** [review]
@shyam-patel-kira great work! made some comments, but no change is requested so far.

Would appreciate the replies :muscle:

File: apps/contracts/contracts/Zengo.sol (hunk 2 above)
Comment (shyam-patel-kira) [reply apps/contracts/contracts/Zengo.sol:34]: no total vote of that iteration

File: apps/contracts/contracts/Zengo.sol (hunk 3 above)
Comment (shyam-patel-kira) [reply apps/contracts/contracts/Zengo.sol:35]: this vote has nothing to do with quadratic voting...this is only for the verification stage

File: apps/contracts/contracts/Zengo.sol (hunk 4 above)
Comment (shyam-patel-kira) [reply apps/contracts/contracts/Zengo.sol:105]: Yes

**Review by HabacucMX — APPROVED** [review]

File: apps/contracts/contracts/Zengo.sol (hunk 1 above)
Comment (shyam-patel-kira) [reply apps/contracts/contracts/Zengo.sol:37]: I shall change the mapping as well for the vote

**Pull request merged**